internal/store/leaf/fsck.go

```diff
@@ -59,10 +63,15 @@ func (s *Store) fsckCheckEntry(ctx context.Context, name string) error {
 	// make sure we can actually decode this secret
 	// if this fails there is no way we could fix this
 	if IsFsckDecrypt(ctx) {
-		_, err := s.Get(ctx, name)
+		// we need to make sure Parsing is enabled in order to parse old Mime secrets
+		ctx = ctxutil.WithShowParsing(ctx, true)
+		secret, err := s.Get(ctx, name)
 		if err != nil {
 			return errors.Wrapf(err, "failed to decode secret %s: %s", name, err)
 		}
+		if kv, ok := secret.(*secrets.KV); ok && kv.FromMime() {
```

Comment:
Actually we could have type checked against a locally defined interface `type fromMime interface { FromMime() bool }` here, but let's leave it as is for now.